src/components/ChecWysiwyg.vue

```diff
@@ -0,0 +1,571 @@
+      slotWidth: 0,
+      hasSlot: false,
+      editor: new Editor({
+        extensions: [
+          new Blockquote(),
+          new BulletList(),
+          new Heading({ levels: [1, 2, 3] }),
+          new Bold(),
+          new Italic(),
+          new ListItem(),
+        ],
+        content: '<p>This is just a boring paragraph</p>',
+      }),
+      nonInputListeners,
+    };
+  },
+  computed: {
+    sharedInputProps() {
+      const {
+        additionalInputAttributes,
+        value,
+        variant,
+        id,
+        $attrs,
+      } = this;
+
+      return {
+        ...$attrs,
+        ...additionalInputAttributes,
+        placeholder: this.hasPlaceholder ? this.placeholder : ' ',
+        class: ['wysiwyg__input', this.innerInputClass, {
+          'wysiwyg__input--has-placeholder': this.hasPlaceholder,
+        }],
+        disabled: variant === 'disabled',
+        id,
+        ref: 'input',
+        value,
+      };
+    },
+    classNames() {
+      const {
+        currency,
+        label,
+        value,
+        styleVariant,
+        variant,
+      } = this;
+
+      return {
+        'wysiwyg--currency': currency,
+        'wysiwyg--disabled': variant === 'disabled',
+        'wysiwyg--error': variant === 'error',
+        'wysiwyg--empty': value === '',
+        'wysiwyg--inline-label': label,
+        'wysiwyg--modified': label ? !!value : false,
+        [`wysiwyg--${styleVariant}`]: styleVariant !== '',
+      };
+    },
+    hasPlaceholder() {
+      return typeof this.placeholder === 'string' && this.placeholder.length > 0;
+    },
+  },
+  watch: {
+    value() {
+      this.$nextTick(() => {
+        this.autoGrow();
+      });
+    },
+    isScrollable(scrollable) {
+      if (scrollable) {
+        this.$refs.input.addEventListener('scroll', this.handleScroll);
+      } else {
+        this.$refs.input.removeEventListener('scroll', this.handleScroll);
+      }
+    },
+  },
+  mounted() {
+    this.autoGrow();
+
+    if (!this.multiline && this.$slots.default) {
+      this.slotObserver = new MutationObserver(this.adjustSlotWidth);
+
+      this.slotObserver.observe(
+        this.$refs.rightContentSlot,
+        {
+          attributes: true,
+          childList: true,
+          characterData: true,
+          subtree: true,
+        },
+      );
+
+      this.adjustSlotWidth();
+    }
+  },
+  beforeDestroy() {
+    this.editor.destroy();
+  },
+  methods: {
+    handleInput($event) {
+      /**
+       * Emitted when the `<input>`'s 'input' event bubbles up. The v-model directive uses this to function.
+       * @event input
+       * @type {$event.target.value}
+       */
+      this.autoGrow();
+      this.$emit('input', $event.target.value, $event);
+    },
+    adjustSlotWidth() {
+      this.hasSlot = this.$refs.rightContentSlot.clientWidth > 0;
+      this.slotWidth = this.$refs.rightContentSlot.scrollWidth;
+    },
+    autoGrow() {
+      if (!this.multiline) {
```

Comment:
Is this a copy paste from TextField? I assume WYSIWYG editors would always be multiline